pkg/controller/storagecluster/cephcluster.go

```diff
@@ -519,25 +520,30 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 // The checkTuneStorageDevices function checks whether devices from the given
 // storage class are a known type that should expclitly be tuned for fast or
 // slow access.
-func (r *ReconcileStorageCluster) checkTuneStorageDevices(storageClassName string) (diskSpeed, error) {
-	storageClass := &storagev1.StorageClass{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: storageClassName}, storageClass)
-	if err != nil {
-		return diskSpeedUnknown, fmt.Errorf("failed to retrieve StorageClass %q. %+v", storageClassName, err)
-	}
-
-	for _, dt := range knownDiskTypes {
-		if string(dt.provisioner) != storageClass.Provisioner {
-			continue
+func (r *ReconcileStorageCluster) checkTuneStorageDevices(storageClassName string, deviceType string) (diskSpeed, error) {
```

Comment:
I was originally thinking of handing the `ds` (deviceset) in instead of the various derived entities.
Then inside the function you can do something like:
```
   storageClassName := ds.DataPVCTemplate.Spec.StorageClassName
   deviceType := ds.DeviceType
```
What do you think?

Reply:
yes even better.